### DISCUSSION TITLE
Cannot print individual courses

### DISCUSSION THREAD

**Discussion by chazmorton** [ucfopen/canvasapi#604] [Q&A] [answered]
While trying to print an individual course from the PaginatedList of course, I am getting the error:
"AttributeError: 'Course' object has no attribute 'course_code'"

I have my API_URL and API_KEY but do not want to show in the code. 

```
canvas = Canvas(API_URL, API_KEY)

user = canvas.get_current_user()

courses = user.get_courses()

for course in courses:
    print(course)
```

I am wondering if anyone knows how to fix this problem. I am following the ucfopen README but their code for going over a PaginatedList of courses doesn't work for me. Thanks.

**Comment by bennettscience** [comment]
That's a strange bug - when you try to print the `Course` object, it looks for `course_code` by default. Since you have one without that code, it fails.

I'll get a PR up tomorrow to fix it, but in the meantime, you can do this:

```python
canvas = Canvas(API_URL, API_KEY)
user = canvas.get_current_user()
courses = user.get_courses()
for course in courses:
    print(course.name, course.id)
```

That'll let you find the course you need along with its ID.

**Reply by chazmorton** [reply]
When I try to print course.name, course.id it gives this error:

```
File "/user/opt/anaconda3/lib/python3.9/site-packages/canvasapi/canvas_object.py", line 14, in __getattribute__
    return super(CanvasObject, self).__getattribute__(name)
AttributeError: 'Course' object has no attribute 'name'
```

**Reply by bennettscience** [reply]
Ok, that's really weird too. Can you try `len(list(courses))` to make sure there are results to loop over? If there are, you could try `courses[0].__dict__` just to see what's available in that first result.

**Reply by chazmorton** [reply]
``` len(list(courses)) ``` did work, it printed out 33. When I did ``` courses[0].__dict__ ``` it gave me ``` {'_requester': <canvasapi.requester.Requester object at 0x7fd060256529>, 'id': 69776, 'access_restricted_by_date': True} ```. It seems like there are a lot of old courses within this list that are not available to me. I tried to ``` get_course(69776) ``` although it wouldn't let me. 

However, doing ``` courses[32].__dict__ ``` it gave me one of my current courses, from which i was able to access the name by doing ``` get_course(32) ```. It seems like my current courses are in there but scrambled about the locked courses. I will just have to filter through them to get ones where ``` access_restricted_by_date != True ```. 

Although it isn't ideal, it is still a working solution! Thanks!

**Reply by bennettscience** [reply, marked as answer]
Ah, that makes sense. I knew Canvas would restrict access to courses, but I didn't know it didn't send _any_ information if it was restricted.

I'm pretty sure using the `enrollment_state` keyword argument will filter out classes you don't have access to any more because of date limits: 

```python
canvas = Canvas(API_URL, API_KEY)
user = canvas.get_current_user()
courses = user.get_courses(enrollment_state="active") 
for course in courses:
    print(course)
```

If that doesn't work, you can try the `state="active"` argument.
